packages/plugin-core/src/commands/OpenLink.ts

```diff
@@ -25,47 +26,106 @@ export class OpenLinkCommand extends BasicCommand<CommandOpts, CommandOutput> {
   async execute() {
     const ctx = DENDRON_COMMANDS.OPEN_LINK;
     this.L.info({ ctx });
+
+    let text = "";
+
     if (!isAnythingSelected()) {
-      const error = DendronError.createFromStatus({
-        status: ERROR_STATUS.INVALID_STATE,
-        message: `nothing selected`,
-      });
-      this.L.error({ error });
-      return { error };
+      window.showInformationMessage(
+        "nothing selected, searching for valid link"
+      );
+    }
+
+    const editor = VSCodeUtils.getActiveTextEditor();
+
+    if (editor) {
+      const docText = editor.document.getText();
+      const offsetStart = editor.document.offsetAt(editor.selection.start);
+      const offsetEnd = editor.document.offsetAt(editor.selection.end);
+
+      const selectUri = true;
```

Comment:
instead of parsing the link in here, can you add a method `getURLAt` in [utils/editor](https://github.com/dendronhq/dendron/blob/21689914d0c84735d243b988dcceb276df97380f/packages/plugin-core/src/utils/editor.ts#L122-L122)?
you can take a look at [CopyLinkUrl](https://github.com/dendronhq/dendron/blob/fe83a99d11feffb0c588a5ae3b23b73b6283c851/packages/plugin-core/src/commands/CopyNoteURL.ts#L62-L62) to see how we use this on a link